fsm/src/main/java/org/egov/fsm/repository/querybuilder/FSMQueryBuilder.java

```diff
@@ -0,0 +1,151 @@
+package org.egov.fsm.repository.querybuilder;
+
+import java.util.List;
+import org.egov.fsm.config.FSMConfiguration;
+import org.egov.fsm.web.model.FSMSearchCriteria;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+@Component
+public class FSMQueryBuilder {
+
+	@Autowired
+	private FSMConfiguration config;
+				
+			private static final String Query = "select fsm.*,fsm_address.*,fsm_geo.*,fsm_pit.*,fsm.id as fsm_id, fsm.createdby as fsm_createdby,"
+					+ " fsm.lastmodifiedby as fsm_lastmodifiedby, fsm.createdtime as fsm_createdtime, fsm.lastmodifiedtime as fsm_lastmodifiedtime,"
+					+ "	 fsm.additionaldetails,fsm_address.id as fsm_address_id,fsm_geo.id as fsm_geo_id,"
+					+ "	 fsm_pit.id as fsm_pit_id"
+					+ "	 FROM eg_fsm_application fsm"
+					+ "	 INNER JOIN   eg_fsm_address fsm_address on fsm_address.fsm_id = fsm.id"
+					+ "	 LEFT OUTER JOIN  eg_fsm_geolocation fsm_geo on fsm_geo.address_id = fsm_address.id"
+					+ "	 LEFT OUTER JOIN  eg_fsm_pit_detail fsm_pit on fsm_pit.fsm_id = fsm.id";
+			
+	private final String paginationWrapper = "SELECT * FROM "

```

Comment:
We don't want to use dense rank as it gives performance issues. We need to rewrite the query

Reply:
taken care